src/transform/src/metadata.ts

```diff
@@ -40,6 +41,24 @@ export class HypermodeMetadata {
     this.functions.push(...functions);
   }
 
+  addEmbedders(functions: FunctionSignature[]) {
+    for (const func of functions) {
+      if (func.parameters.length != 1)
+        throw new Error(
+          "Expected embedder to have one argument, but found " +
+            (func.parameters.length ? "multiple" : "no") +
+            " arguments!",
+        );
+      if (func.parameters[0].type.name != "string")
```

Comment:
to future proof this, could you change it to support string[] as well, we want to allow batching. And separately, can it also check that the return type is f64[] or f64[][]

Reply:
Yeah, let's support two signatures.  Either `string => f64[]` or `string[] => f64[][]`

Reply:
Sure

Reply:
@jairad26 @mattjohnsonpint resolved.